### ISSUE TITLE
debug:container --types & debug:autowiring can fail with missing deps

### ISSUE THREAD

**Issue by weaverryan** [symfony/symfony#24639]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.3 (I believe)

Here's the setup:

1) Use Flex
2) Install SensioFrameworkExtraBundle
3) Run `bin/console debug:container --types`

<img width="1279" alt="screen shot 2017-10-19 at 7 26 28 pm" src="https://user-images.githubusercontent.com/121003/31798760-988f25ac-b503-11e7-814f-50aacaa0aa76.png">

The issue is that SensioFrameworkExtraBundle registers a service that relies on `symfony/security`. Since security is not present, this service is removed: https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/master/DependencyInjection/Compiler/OptimizerPass.php#L27

However, I believe that the `debug:container` command uses the container before some of these removal passes. So, this service is still present when we check to see if the class exists: https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php#L248. The autoloader causes the class to be loaded, and error because the parent class is missing.

So... long way of saying, we just need a safer way of checking to see if the service id is an existent class.

**Comment by nicolas-grekas** [comment]
What about *not* checking that (but checking that there is a `\` in the id instead?) Would the list be that different?

**Comment by weaverryan** [comment]
Great question! For a Flex app with some basic dependencies installed, the *only* alias that does not have a `\` in it is `Twig_Environment`... which of course also has a `Twig\Environment` alias.

So in practice, if we switched to look for a `\`... that would fix it. But there would be edge-case libraries where we would be in-accurate.

What if we did:

A) If there is a `\`, then we immediately assume it is a valid type (without calling `class_exists()`)
B) if not, we use `class_exists`. Or better, we use reflection with a try/catch (like `AutowirePass`)
